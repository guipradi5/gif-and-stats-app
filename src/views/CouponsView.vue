<template>
  <main>
    <h1>Coupons</h1>

    <section v-if="couponsData" class="coupons-stats">
      <div>There's a total of {{couponsData.count}} coupons</div>
      <div>
        <h3>For the "percent-off" coupons, we have the following statistics:</h3>
        <ul>
          <li>Count: {{couponsData.typeCount.percentOff}}</li>
          <li>Minimum Discount: {{couponsData.percentOffData.minValue}}%</li>
          <li>Maximum Discount: {{couponsData.percentOffData.maxValue}}%</li>
          <li>Average Discount: {{couponsData.percentOffData.average}}%</li>
        </ul>
      </div>
      <div>
        <h3>For the "dollar-off" coupons, we have the following statistics:</h3>
        <ul>
          <li>Count: {{couponsData.typeCount.dollarOff}}</li>
          <li>Minimum Discount: {{couponsData.dollarOffData.minValue}}$</li>
          <li>Maximum Discount: {{couponsData.dollarOffData.maxValue}}$</li>
          <li>Average Discount: {{couponsData.dollarOffData.average}}$</li>
        </ul>
      </div>
      
      <div>
        <h3>For each retailer here are the following statistics:</h3>
        <div v-for="retailer in couponsData.retailers" :key="retailer.name">
          <h4>{{retailer.name}}</h4>
          <div>
            <h5>Percent Off Discounts</h5>
            <ul>
              <li>Coupons: {{retailer.percentOff.count}}</li>
              <li>Minimum Discount: {{retailer.percentOff.minValue}}%</li>
              <li>Maximum Discount: {{retailer.percentOff.maxValue}}%</li>
              <li>Average Discount: {{retailer.percentOff.average}}%</li>
            </ul>
          </div>
          <div>
            <h5>Dollar Off Discounts</h5>
            <ul>
              <li>Coupons: {{retailer.dollarOff.count}}</li>
              <li>Minimum Discount: {{retailer.dollarOff.minValue}}%</li>
              <li>Maximum Discount: {{retailer.dollarOff.maxValue}}%</li>
              <li>Average Discount: {{retailer.dollarOff.average}}%</li>
            </ul>
          </div>
        </div>
      </div>
    </section>
    <section class="coupons-graph">
    </section>
  </main>
</template>

<script setup>
import { getCouponsData } from "@/api/api";
import { onMounted, ref } from "vue";

const couponsCount = ref(0)
const couponsData = ref(null)

onMounted(async () => {
  couponsData.value = await getCouponsData()
})

</script>


<style scoped lang="scss">
main {
  padding: 0 30px;
  h3, h4, h5{
    margin-bottom: 0.3em;
  }
}
</style>
