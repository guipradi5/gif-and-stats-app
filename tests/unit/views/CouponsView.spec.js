import { shallowMount } from "@vue/test-utils";
import CouponsView from "@/views/CouponsView.vue";

jest.mock("@/api/api", () => ({
  ...jest.requireActual("@/api/api"),
  getCouponsData: () => {
    return {
      count: 3,
      coupons: ["coupon"],
      dollarOffData: { minValue: 55, maxValue: 234, average: 6 },
      percentOffData: { minValue: 1, maxValue: 5, average: 8 },
      retailers: {
        macys: {
          dollarOff: { average: "17.22", count: 9, maxValue: 50, minValue: 10 },
          name: "macys",
          percentOff: {
            average: "25.00",
            count: 1,
            maxValue: 25,
            minValue: 25,
          },
        },
        nike: {
          dollarOff: {
            average: "75.00",
            count: 2,
            maxValue: 150,
            minValue: 150,
          },
          name: "nike",
          percentOff: { average: "0.00", count: 1, maxValue: 0, minValue: 0 },
        },
        nordstrom: {
          dollarOff: {
            average: "75.00",
            count: 3,
            maxValue: 150,
            minValue: 75,
          },
          name: "nordstrom",
          percentOff: { average: "0.00", count: 1, maxValue: 0, minValue: 0 },
        },
      },
      typeCount: {
        buyOneGetOne: "3",
        dollarOff: "5",
        freeGift: "7",
        freeShipping: "9",
        percentOff: "4",
      },
    };
  },
}));

const component = () => {
  return shallowMount(CouponsView, {});
};

describe("CouponsView.vue", () => {
  it("CouponsView should mount", () => {
    const wrapper = component();
    expect(wrapper.vm).toBeTruthy();
  });

  it("CouponsView should render", async () => {
    const wrapper = await component();
    await wrapper.vm.$nextTick();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
