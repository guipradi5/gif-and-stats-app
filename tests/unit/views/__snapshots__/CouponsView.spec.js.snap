// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CouponsView.vue CouponsView should render 1`] = `
<main>
  <h1>Coupons</h1>
  <section class="coupons-stats">
    <div>There's a total of 3 coupons</div>
    <div>
      <h3> For the "percent-off" coupons, we have the following statistics: </h3>
      <ul>
        <li>Count: 4</li>
        <li>Minimum Discount: 1%</li>
        <li>Maximum Discount: 5%</li>
        <li>Average Discount: 8%</li>
      </ul>
    </div>
    <div>
      <h3>For the "dollar-off" coupons, we have the following statistics:</h3>
      <ul>
        <li>Count: 5</li>
        <li>Minimum Discount: 55$</li>
        <li>Maximum Discount: 234$</li>
        <li>Average Discount: 6$</li>
      </ul>
    </div>
    <div>
      <h3>For each retailer here are the following statistics:</h3>
      <div>
        <h4>macys</h4>
        <div>
          <h5>Percent Off Discounts</h5>
          <ul>
            <li>Coupons: 1</li>
            <li>Minimum Discount: 25%</li>
            <li>Maximum Discount: 25%</li>
            <li>Average Discount: 25.00%</li>
          </ul>
        </div>
        <div>
          <h5>Dollar Off Discounts</h5>
          <ul>
            <li>Coupons: 9</li>
            <li>Minimum Discount: 10%</li>
            <li>Maximum Discount: 50%</li>
            <li>Average Discount: 17.22%</li>
          </ul>
        </div>
      </div>
      <div>
        <h4>nike</h4>
        <div>
          <h5>Percent Off Discounts</h5>
          <ul>
            <li>Coupons: 1</li>
            <li>Minimum Discount: 0%</li>
            <li>Maximum Discount: 0%</li>
            <li>Average Discount: 0.00%</li>
          </ul>
        </div>
        <div>
          <h5>Dollar Off Discounts</h5>
          <ul>
            <li>Coupons: 2</li>
            <li>Minimum Discount: 150%</li>
            <li>Maximum Discount: 150%</li>
            <li>Average Discount: 75.00%</li>
          </ul>
        </div>
      </div>
      <div>
        <h4>nordstrom</h4>
        <div>
          <h5>Percent Off Discounts</h5>
          <ul>
            <li>Coupons: 1</li>
            <li>Minimum Discount: 0%</li>
            <li>Maximum Discount: 0%</li>
            <li>Average Discount: 0.00%</li>
          </ul>
        </div>
        <div>
          <h5>Dollar Off Discounts</h5>
          <ul>
            <li>Coupons: 3</li>
            <li>Minimum Discount: 75%</li>
            <li>Maximum Discount: 150%</li>
            <li>Average Discount: 75.00%</li>
          </ul>
        </div>
      </div>
    </div>
  </section>
  <section class="coupons-graph"></section>
</main>
`;
